doc-examples/hibernate-example-kotlin-ksp/src/test/kotlin/example/BeanIntrospectionSpec.kt

```diff
@@ -0,0 +1,26 @@
+package example
+
+import io.micronaut.core.annotation.Introspected
+import io.micronaut.core.beans.BeanIntrospectionReference
+import io.micronaut.core.beans.BeanIntrospector
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import jakarta.persistence.Entity
+import org.junit.jupiter.api.Test
+import java.util.*
+
+@MicronautTest(transactional = false)
+class BeanIntrospectionSpec() {
+
+    @Test
+    fun testFindEntities() {
+        val packageName = this.javaClass.packageName
+        val allEntities = BeanIntrospector.SHARED.findIntrospections(Entity::class.java)
+        val allIntrospected = BeanIntrospector.SHARED.findIntrospections(Introspected::class.java)
+        val introspectedInPackage = BeanIntrospector.SHARED.findIntrospections(Introspected::class.java, packageName)
+        val entitiesInPackage = BeanIntrospector.SHARED.findIntrospections(Entity::class.java, packageName)
+        assert(allIntrospected.isNotEmpty())
```

Comment:
This test finds one class `SimpleEntity` annotated with `@Introspected` but none instance of `@Entity` because it seems ksp processor is not annotating `@Entity` classes with `@Introspected`.
@graemerocher don't know what is difference from Kotlin processor, maybe you will know if something is missing?